asf_search/download/download.py

```diff
@@ -93,3 +95,30 @@ def strip_auth_if_aws(r, *args, **kwargs):
         location = r.headers['location']
         r.headers.clear()
         r.headers['location'] = location
+
+def _poll(response: Response, session: ASFSession, url):
+    tries = 0
+    while response.status_code == 202 and tries < 3:
+        response = _try_get_response(session=session, url=url)
+        
+        if response.status_code != 200:
+            tries += 1
+            sleep(10)
+    
+    if response.status_code != 200:
+        raise ASFDownloadError(f'Failed to download BURST product. HTTP {response.status_code}: {response.text}')
+
+    return response
+
+def _try_get_response(session: ASFSession, url: str):
+    response = session.get(url, stream=True, hooks={'response': strip_auth_if_aws})
+
+    try:
+        response.raise_for_status()
+    except HTTPError as e:
+        if 400 <= response.status_code <= 499:
+            raise ASFAuthenticationError(f'HTTP {e.response.status_code}: {e.response.text}')
+
+        raise e
```

Comment:
Just thinking out loud, why not have the poll logic built here directly? Something like:

```python
def _try_get_response(session: ASFSession, url: str, retries: int = 0):
    response = session.get(url, stream=True, hooks={'response': strip_auth_if_aws})

    try:
        response.raise_for_status()
    except HTTPError as e:
        # This block here:
        if response.status_code == 202:
            if retries == 0:
                raise
            else:
                sleep(20)
                return _try_get_response(session: ASFSession, url: str, retries = retries-1)
        # Old stuff:
        elif 400 <= response.status_code <= 499:
            raise ASFAuthenticationError(f'HTTP {e.response.status_code}: {e.response.text}')

        raise e
```
Or something along these lines? It just seems weird to sometimes call _poll, and other times call _try_get_response, just based on if it's a burst. Maybe that's okay though? Food for thought

Reply:
It's recommend we wait about 30 seconds after the first query to the burst extractor if we get a 202 back (and 10 seconds for each subsequent 202, just to be safe), which I think would make this approach a little more complicated with that initial query with the longer wait time.

Reply:
Hmm I don't like how I had retries only trigger with 202's too, since it's a param for the *entire* method. If we end up going this route, it could maybe look something like:

```python
def _try_get_response(session: ASFSession, url: str, retries: int = 0, wait: int = 0):
    response = session.get(url, stream=True, hooks={'response': strip_auth_if_aws})

    try:
        response.raise_for_status()
    except HTTPError as e:
        if retries != 0:
            # It doesn't matter what the status code was if we're set to retry anyway:
            sleep(wait)
            return _try_get_response(session, url, retries = retries-1, wait = 10)

        elif 400 <= response.status_code <= 499:
            raise ASFAuthenticationError(f'HTTP {e.response.status_code}: {e.response.text}')
        raise e
    return response
```
Then the call for bursts could look like this, to give the 30 sec initial and 10 sec followups?
```python
query =  _try_get_response(my_sesh, my_url, retries=3, wait=30)
```
But I think that last draft is about as clean as I can get it. What are your thoughts on this compared to having `_poll` broken out as a second method for bursts?

Reply:
Bear in mind that once the extractor is upgraded with duplicate-job handling, a lot of this will get reduced to just polling once every second.

Reply:
Also, consider Tenacity! https://tenacity.readthedocs.io/en/latest/
as per the Engineering discussion around here: https://chat.asf.alaska.edu/asf/pl/sfwajnhhm3g678gj49q3pgusfe